### ISSUE TITLE
Tucker and CP: show a real downstream application of each, driven by a widget

### ISSUE THREAD

**Issue by project-delphi** [project-delphi/tensors-workshop#18]
## The problem

Section 10 earns its punchline — the hour factor peaks at 18 and the
decomposition found rush hour by itself (`scripts/content.py:1518`). But the
"so what" is one paragraph of prose that nobody can touch:

> **Where this is used.** In tech, Tucker and CP compress the large weight
> tensors inside neural networks so models run on phones instead of servers. In
> biotech, applied to data such as (genes × samples × conditions), they find
> structure ordinary PCA cannot reach…

That is an assertion, not a demonstration. A student leaves able to compute an
HOSVD and still unsure why anyone pays for one. CP is worse off: it exists in
this workshop only as prose plus a commented-out `parafac` call in take-home C
(`scripts/content.py:1704-1725`). Nobody ever sees a CP component.

## The ask

**Show a real downstream use of Tucker and of CP, and make each one a thing the
student drives with a widget.** The interaction is the argument: moving a rank
slider and watching the payoff move is what converts "compression" from a word
into a trade-off you have felt.

Two applications, one per decomposition, each answering "what does this buy
someone who ships things":

- [ ] **Tucker → make a model small enough to run on a phone.** Take one real
      conv layer weight tensor (or a comparable dense tensor), Tucker-compress
      it, and put a **rank slider** on it. The slider drives, live: parameter
      count, compression ratio, reconstruction error, and multiply-add count.
      The point students should reach on their own is that the curve has a knee
      — most of the size goes away long before the error becomes visible.
- [ ] **CP → recover interpretable components nobody labelled.** CP's selling
      point is not accuracy, it is **essential uniqueness**: the components mean
      something, where PCA's rotate freely. Give it a tensor where the answer is
      checkable — the existing taxi tensor (pickup × dropoff × hour) is already
      loaded and its components read as real trip types, so it is the cheap
      option; a fluorescence EEM (excitation × emission × sample) is the classic
      PARAFAC case where the components come out as actual chemical species.
      Widget: a **component selector** that draws the three factor vectors of
      component *r* side by side, so you can flip through them and name each one.

A third, if either of the above is cut for time: **completion.** Knock out a
fraction of the taxi tensor's entries with a slider and show CP/Tucker filling
them back in. That is the recommender-system and sensor-gap story in one control.

Take these as the starting list, not a specification. One well-built widget per
decomposition beats three shallow ones.

## Where this goes

Section 10 has a **15-minute budget** and is already full. Do not spend it.
Options, in order of preference:

1. A new **appendix / take-home** block in section 11, alongside take-home C —
   which should stop being a commented-out `parafac` call and become the CP
   application above.
2. An **optional, clearly-marked** final cell in notebook 10, flagged as "run
   this after the workshop", the same way §04's optional figure is marked.

If any of it is worth putting on the slides, that is a separate PR and it
overlaps with #17 (the factorization-ladder overview slide) — coordinate rather
than duplicating.

## Constraints — read before writing code

- **Edits go in `scripts/content.py`**, never in `notebooks/*.ipynb`. The
  notebooks are generated by `scripts/gen_notebooks.py`; CI reruns both
  generators and fails if the tree changes.
- **No committed outputs.** `scripts/check_links.py` fails on any cell with
  outputs or an execution count. A widget that renders only on execution is
  fine; a widget whose state is serialised into the `.ipynb` on save is not.
  Check `git diff --stat` before committing.
- **`ipywidgets` on Colab** — students are on Colab, not local Jupyter. Some
  widget types need `google.colab.output.enable_custom_widget_manager()` first;
  put that in the cell with a comment rather than assuming the runtime is
  configured. Test in Colab, not only locally. Same constraint as #9, which is
  the umbrella issue for visuals across all twelve notebooks — this issue is the
  §10/§11 depth case, so follow whatever widget idiom #9 lands on.
- **New dependency decision.** Nothing in this repo currently imports
  `tensorly`, and the local run line in `CLAUDE.md` lists only
  numpy/pandas/scikit-learn/scikit-image/scipy/jupyterlab. Either write CP as a
  short ALS loop in plain NumPy (consistent with §10, which builds HOSVD by hand
  from unfold + SVD + einsum) or add a `%pip install -q tensorly` cell and say
  so in the notebook. Do not silently assume it is installed.
- **Keep the data real and small.** Every dataset here downloads at runtime and
  the whole notebook has to stay usable on a free Colab CPU. If the Tucker
  example needs conv weights, they must come from something that downloads in
  seconds — no multi-hundred-MB checkpoint.
- **Any number printed in prose must be one you actually ran.** The plan
  document lists its verified numbers explicitly (taxi Tucker: 4.71×, 6.7%
  error, hour factor peaking at 18); new claims are held to the same bar.
- **Bilingual prose** on every new markdown cell — 🇪🇸 blockquote alongside the
  English, matching the existing pattern.
- **No commits on `main`** — branch, then PR.

```bash
uv run --with pyyaml,nbformat python scripts/gen_notebooks.py
uv run --with pyyaml,nbformat python scripts/check_links.py --notebooks-only
```

## Acceptance criteria

- [ ] One Tucker application and one CP application, each with a working widget,
      each stating in one sentence who benefits and how
- [ ] CP appears somewhere as executable code, not only as prose and a comment
- [ ] Section 10's 15-minute budget is unchanged — new material is take-home or
      explicitly optional
- [ ] Generators rerun and `check_links.py` passes; `git diff --stat` shows no
      output blobs in the `.ipynb`
- [ ] Opened in Colab and the sliders actually move — verified there, not just
      in local Jupyter
- [ ] Bilingual prose on every new markdown cell

## Related

#9 (a plot or widget in every notebook), #17 (overview slide: what different
factorizations buy you), #7 (SVD rank reduction you can see — the matrix-level
version of the same idea).
